Classes/Plugin/OaiPmh.php

```diff
@@ -931,24 +946,32 @@ protected function fetchDocumentUIDs()
     {
-        $documentsToProcess = $documentListSet->removeRange(0, intval($this->conf['limit']));
+        $documentsToProcess = $documentListSet->removeRange(0, (int)$this->conf['limit']);
         $verb = $this->piVars['verb'];
-        $documents = $GLOBALS['TYPO3_DB']->exec_SELECT_mm_query(
-            'tx_dlf_documents.*,GROUP_CONCAT(DISTINCT tx_dlf_collections.oai_name ORDER BY tx_dlf_collections.oai_name SEPARATOR " ") AS collections',
-            'tx_dlf_documents',
-            'tx_dlf_relations',
-            'tx_dlf_collections',
-            'AND tx_dlf_documents.uid IN (' . implode(',', $GLOBALS['TYPO3_DB']->cleanIntArray($documentsToProcess)) . ')'
-                . ' AND tx_dlf_documents.pid=' . intval($this->conf['pages'])
-                . ' AND tx_dlf_collections.pid=' . intval($this->conf['pages'])
-                . ' AND tx_dlf_relations.ident=' . $GLOBALS['TYPO3_DB']->fullQuoteStr('docs_colls', 'tx_dlf_relations')
-                . Helper::whereClause('tx_dlf_collections'),
-            'tx_dlf_documents.uid',
-            'tx_dlf_documents.tstamp',
-            $this->conf['limit']
-        );
+
+        $connection = GeneralUtility::makeInstance(ConnectionPool::class)->getConnectionForTable('tx_dlf_documents');
+
+        $sql = 'SELECT `tx_dlf_documents`.*, GROUP_CONCAT(DISTINCT `tx_dlf_collections`.oai_name ORDER BY tx_dlf_collections.oai_name SEPARATOR " ") AS `collections` ' .
+            'FROM `tx_dlf_documents` ' .
+            'INNER JOIN `tx_dlf_relations` ON `tx_dlf_relations`.`uid_local` = `tx_dlf_documents`.`uid` ' .
+            'INNER JOIN `tx_dlf_collections` ON `tx_dlf_collections`.`uid` = `tx_dlf_relations`.`uid_foreign` ' .
+            'WHERE (((`tx_dlf_documents`.`deleted` = 0) AND (`tx_dlf_collections`.`deleted` = 0)) AND ((`tx_dlf_documents`.`hidden` = 0) AND (`tx_dlf_collections`.`hidden` = 0)) AND (`tx_dlf_documents`.`starttime` <= ' . $GLOBALS['EXEC_TIME'] . ') AND ((`tx_dlf_documents`.`endtime` = 0) OR (`tx_dlf_documents`.`endtime` > ' . $GLOBALS['EXEC_TIME'] . '))) ' .
+            'AND tx_dlf_documents.uid IN ( ? ) ' .
+            'AND tx_dlf_documents.pid = ? ' .
+            'AND tx_dlf_collections.pid = ? ' .
+            'AND tx_dlf_relations.ident="docs_colls" ' .
+            Helper::whereClause('tx_dlf_collections') .
+            ' LIMIT ?';
+
+        $documents = $connection->prepare($sql);
+        $documents->bindValue(1, implode(',', $documentsToProcess));
+        $documents->bindValue(2, (int)$this->conf['pages']);
+        $documents->bindValue(3, (int)$this->conf['pages']);
+        $documents->bindValue(4, (int)$this->conf['limit']);
+        $documents->execute();
+
         $output = $this->oai->createElementNS('http://www.openarchives.org/OAI/2.0/', $verb);
-        while ($resArray = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($documents)) {
+        while ($resArray = $documents->fetch()) {
```

Comment:
Actually the $resArray() is an array with only `NULL` values. So the query above must be wrong on calling 
 
[verb=ListRecords&metadataPrefix=mets](url)

<pre>
array(41 items)
   uid => NULL
   pid => NULL
   tstamp => NULL
   crdate => NULL
   cruser_id => NULL
   deleted => NULL
   hidden => NULL
   starttime => NULL
   endtime => NULL
   fe_group => NULL
   prod_id => NULL
   location => NULL
   record_id => NULL
   opac_id => NULL
   union_id => NULL
   urn => NULL
   purl => NULL
   title => NULL
   title_sorting => NULL
   author => NULL
   year => NULL
   place => NULL
   thumbnail => NULL
   metadata => NULL
   metadata_sorting => NULL
   structure => NULL
   partof => NULL
   volume => NULL
   volume_sorting => NULL
   license => NULL
   terms => NULL
   restrictions => NULL
   out_of_print => NULL
   rights_info => NULL
   collections => NULL
   mets_label => NULL
   mets_orderlabel => NULL
   owner => NULL
   solrcore => NULL
   status => NULL
   document_format => NULL
</pre>

The same with

[?verb=ListIdentifiers&metadataPrefix=oai_dc](url)
[?verb=ListIdentifiers&metadataPrefix=mets](url)
[?verb=ListIdentifiers&metadataPrefix=epicur](url)
[?verb=ListRecords&metadataPrefix=oai_dc](url)
[?verb=ListRecords&metadataPrefix=epicur](url)

Reply:
This is partly fixed. The array contains values. But the query is still somehow wrong. There is always only one document found.


**Example**

`$documentsToProcess` containts the UIDs 1 to 10.

```
SELECT `tx_dlf_documents`.*, GROUP_CONCAT(DISTINCT `tx_dlf_collections`.`oai_name` ORDER BY `tx_dlf_collections`.`oai_name` SEPARATOR " ") AS `collections`              FROM `tx_dlf_documents`              INNER JOIN `tx_dlf_relations` ON `tx_dlf_relations`.`uid_local` = `tx_dlf_documents`.`uid`              INNER JOIN `tx_dlf_collections` ON `tx_dlf_collections`.`uid` = `tx_dlf_relations`.`uid_foreign`              WHERE `tx_dlf_documents`.`uid` IN ( 1,2,3,4,5,6,7,8,9,10 ) AND `tx_dlf_relations`.`ident`="docs_colls";
```
**Result (shorted)**

```
*************************** 1. row ***************************
             uid: 1
             pid: 12
          tstamp: 1569320630
          crdate: 1564134023
       cruser_id: 2
         deleted: 0
          hidden: 0
       starttime: 0
         endtime: 0
        fe_group: 
         prod_id: 336246
        location: https://example.com/00052637/00052637_mets.xml
       record_id: 00052637
         opac_id: 
        union_id: 
           title: Title of Document
          author: 
            year: 
           place: 
       structure: 33
          partof: 0
          volume: 
  volume_sorting: 
     collections: 1
           owner: 1
        solrcore: 1
          status: 0
 document_format: METS
         license: 
           terms: 
    restrictions: 
    out_of_print: 
     rights_info: 
      mets_label: 
 mets_orderlabel: 

1 row in set (0.001 sec)

```